### PR TITLE
download jars instead of bundling them

### PR DESCRIPTION
**Pull request by pb82**
Plugins are now supplied as URLs and downloaded by the init container instead of bundling them in the image.

Themes are bundled and copied as usual.

### CODE REVIEW

File: plugins.py [hunk 1]
```diff
@@ -1,34 +1,64 @@
...
+import random
+import string
+from shutil import copyfileobj
 from distutils.dir_util import copy_tree
 from distutils import log
 
-pluginsKey = "SSO_PLUGINS"
-pluginsVolume = "/opt/plugins"
+extensionsKey = "KEYCLOAK_EXTENSIONS"
+extensionsVolume = "/opt/extensions"
```
Comment (davidffrench) [inline plugins.py:11]: Was it mentioned that the path for RH-SSO was different to Keycloak?

File: plugins.py [hunk 2]
```diff
@@ -1,34 +1,64 @@
...
-    pluginPath = './' + plugin + '.jar'
-    if os.path.isfile(pluginPath):
-        print("Copying plugin: " + plugin)
-        copy(pluginPath, pluginsVolume)
+# Try to get the filename from the response headers and return
+# a random name if that fails
+def parseHeader(headers):
+    header = headers["Content-Disposition"]
+    if not header:
+        return randomFileName()
```
Comment (davidffrench) [inline plugins.py:22]: Could you explain why if this header doesn't exist that a random filename will be generated? Does this header always contain the file name?

**Review by davidffrench — COMMENTED** [review]
Great work @pb82 , this work is going to be hugely beneficially to the Keycloak operator. Just two small comments from me

Do you plan to migrate this repo over to the aerogear org after the changes have been made?

File: plugins.py (hunk 1 above)
Comment (pb82) [reply plugins.py:11]: Yes, the path for RHSSO is different but that is only relevant on the operator side. The correct volume gets always mounted to the init container under this path.

File: plugins.py (hunk 2 above)
Comment (pb82) [reply plugins.py:22]: All file downloads should populate the `Content-Disposition` header. This is just an extra measure to make sure we can fetch the file even if the header is missing for some reason.

File: plugins.py (hunk 1 above)
Comment (davidffrench) [reply plugins.py:11]: ok, thanks @pb82

File: plugins.py (hunk 2 above)
Comment (davidffrench) [reply plugins.py:22]: sounds good, thank you

**Review by davidffrench — APPROVED** [review]
Excellent work, LGTM @pb82

**Pull request closed without merging**